sentry_streams/sentry_streams/pipeline/msg_parser.py

```diff
@@ -31,7 +33,18 @@ def msg_parser(msg: PyRawMessage) -> Any:
     return decoded
 
 
-def msg_serializer(msg: Message[Any]) -> bytes:
+def msg_serializer(msg: Message[Any], dt_format: Optional[str] = None) -> bytes:
     payload = msg.payload
 
-    return json.dumps(payload).encode("utf-8")
+    def custom_serializer(obj: Any, dt_format: Optional[str] = None) -> str:
+        if isinstance(obj, datetime):
+            if dt_format:
+                try:
+                    return obj.strftime(dt_format)
+                except Exception as e:
+                    raise ValueError(f"Invalid datetime format '{dt_format}': {e}") from e
```

Comment:
You do not need to add a Try catch if the only thing the `except` block does is to re-raise the exception without adding anything. Just let the exception bubble up.